lib/metrics/context.js

```diff
@@ -57,8 +62,20 @@ module.exports = function (log, config) {
 
     metadata.service = this.payload.service || this.query.service
 
+    let key
+
     return P.resolve()
-      .then(() => cache.set(getKey(token), metadata))
+      .then(() => {
+        key = getKey(token)
+        return cache.get(key)
```

Comment:
The memcached package has an `add` method that works like `set` but will error out if the key already exists; memcache is fast but it may still be worth exposing and using that to avoid two round-trips from a `get` and a `set` here.

Reply:
> maybe we can replicate something like mozilla/fxa-content-server#6258 in a dev environment

Thanks, I was really trying to find out what those steps to reproduce actually are, but that was pretty lazy (and wrong) of me. I'll have a play around and reproduce it without the fix first, then make sure the fix does what it's supposed to.

Reply:
> The memcached package has an `add` method that works like `set` but will error out if the key already exists; memcache is fast but it may still be worth exposing and using that to avoid two round-trips from a `get` and a `set` here.

Hmm, one slight problem with that is that is reliably distinguishing between an error setting the value because it already exists and an error setting the value because of something else. In the first case we want the call to `stash` to succeed, because we don't want that type of error to fail the wider request.

I could compare `err.message` but it always feels a little bit risky to predicate behviour on implementation details like that. Another approach would be to just ignore all errors from `stashMetricsContext` instead, maybe that's fine because it's not core functionality to the substance of the request.

Reply:
(or rather ignore all errors from `memcached.add`, not from `request.stashMetricsContext`)